### PR TITLE
WALLETS_CREATE_08 - address_pool_gap integration tests

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

# Overview

- [ ] I have added `address_pool_gap` integration tests that somehow did not make it on #194

# Comments

### CODE REVIEW

File: lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs [hunk 1]
```diff
@@ -520,6 +522,184 @@ spec = do
...
+                "address_pool_gap": 132323000
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "An address pool gap must be a natural\
+              \ number between 10 and 100."
+            ]
+
+    it "WALLETS_CREATE_08 - -1000 as address_pool_gap -> fail" $ \ctx -> do
```
Comment (piotr-iohk) [inline lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs:609]: This one, along with the next 3, will fail until #204 is fixed.

**Comment by KtorZ** [comment]
I'll look into fixing the failing cases :thinking:

File: lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs [hunk 2]
```diff
@@ -520,6 +522,182 @@ spec = do
...
+        forM_ matrix $ \(title, addrPoolGap, expectations) -> it title $ \ctx -> do
+            let payload = Json [json| {
+                    "name": "Secure Wallet",
+                    "mnemonic_sentence": #{mnemonics24},
+                    "passphrase": "Secure passphrase",
+                    "address_pool_gap": #{addrPoolGap}
+                    } |]
+            r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+            verify r expectations
+
```
Comment (KtorZ) [inline lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs:566]: I wonder why the tests below aren't part of a matrix like above? :thinking: 
What's the rational for not putting them as part of the matrix?

**Review by KtorZ — APPROVED** [review]

File: lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs (hunk 2 above)
Comment (piotr-iohk) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs:566]: Yes, so GHC was complaining that it is not in range of Word8 = [0...255] ¯\_(ツ)_/¯

Comment (KtorZ) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs:566]: Hehe. Yeah, gotta use `Integer` here. You probably don't need `Word8`

**Pull request merged**